HACKING.md

```diff
@@ -3,51 +3,94 @@
 First, create the Jenkins base infra:
 
 ```
-oc new-app --file=manifests/jenkins.yaml \
-  --param=STORAGE_CLASS_NAME=ocs-storagecluster-ceph-rbd
+oc process -l app=coreos-ci \
+    -p ENABLE_OAUTH=false \
+    -p NAMESPACE=coreos-ci \
+    -p STORAGE_CLASS_NAME=ocs-storagecluster-ceph-rbd \
+    -f https://raw.githubusercontent.com/coreos/fedora-coreos-pipeline/main/manifests/jenkins.yaml | oc create -f -
 ```
 
-Use the `NAMESPACE` parameter if you're not targeting one
+Change the `NAMESPACE` parameter if you're not targeting one
 named `coreos-ci`.
 
-Create the JCasC configmap:
+We turn off the default OpenShift authentication because we
+will use GitHub instead.
+
+### Paste the GitHub OAuth secrets
+
+The GitHub OAuth plugin does not support Jenkins credentials
+so we directly substitute the secret into the JCasC
+configmap. The secret is available in BitWarden.
+
+```
+CLIENT_ID=<SECRET>
+CLIENT_SECRET=<SECRET>
+sed -i -e "s,GITHUB_OAUTH_CLIENT_ID,${CLIENT_ID}," jenkins/config/github-oauth.yaml
+sed -i -e "s,GITHUB_OAUTH_CLIENT_SECRET,${CLIENT_SECRET}," jenkins/config/github-oauth.yaml
+```
+
+### Create the JCasC configmap
 
 ```
 oc create configmap jenkins-casc-cfg --from-file=jenkins/config
 ```
 
-Create the `github-oauth` secret using `oc create -f`. The
-manifest is available in BitWarden.
+### Create the GitHub webhook shared secret
 
-Create the `github-webhook-shared-secret` secret using `oc
-create -f`. The manifest is available in BitWarden.
+The current webhook shared secret used is in BitWarden:
+
+```
+# use `echo -n` to make sure no newline is in the secret
+echo -n "$SECRET" > secret
```

Comment:
The steps in the pipeline say to use `uuidgen -r` here. Are we telling people to pull from bitwarden because we want to share this value amongst the two?

Reply:
It's a different secret from the pipeline's. We could put `uuigden -r` here, but reusing the same webhook secret means we don't have to re-register all the hooks, which is nice. We could consider doing the same for the pipeline.